src/MongoDB.Driver.Core/Core/Misc/DnsClientWrapper.cs

```diff
@@ -38,6 +43,30 @@ public DnsClientWrapper()
         public LookupClient LookupClient => _lookupClient;
 
         // public methods
+        public string GetCanonicalizedHostName(string hostNameOrAddress)
+        {
+#if NETSTANDARD1_5 || NETSTANDARD1_6
+            var hostEntry = Dns.GetHostEntryAsync(hostNameOrAddress).GetAwaiter().GetResult();
+#else
+            var hostEntry = Dns.GetHostEntry(hostNameOrAddress);
+#endif
+            if (hostEntry == null)
+            {
+                return null;
+            }
+
+            var ipAddress = hostEntry.AddressList?.Length > 0 ? hostEntry.AddressList[0] : null;
```

Comment:
Similar to how it's done inside java:

    private static InetAddress getByName(String host, InetAddress reqAddr)
        throws UnknownHostException {
        return InetAddress.getAllByName(host, reqAddr)[0];
    }